### PR TITLE
Content page js to ts and flow Reference

### PR DESCRIPTION
**Pull request by ishouty**

### CODE REVIEW

File: common/views/components/ContentPage/ContentPage.tsx [hunk 1]
```diff
@@ -1,45 +1,41 @@
...
+  Header: ReactElement<typeof PageHeader>;
+  Body: any;
   // This is used for content type specific components e.g. InfoBox
-  children?: ?Node,
-  contributorProps?: ElementProps<typeof Contributors>,
-  Siblings?: Element<typeof SeriesNavigation>[],
-  outroProps?: ?ElementProps<typeof Outro>,
-  seasons?: any, // TODO
-|};
+  children?: ReactNode | null | undefined;
```
Comment (jamesgorrie) [inline common/views/components/ContentPage/ContentPage.tsx:33]: The type of `ReactNode` is `ReactNode = ReactChild | ReactFragment | ReactPortal | boolean | null | undefined`.

So you shouldn't have to have all these type. You should actually be able to have `children: ReactNode` as far as I can tell.

File: common/views/components/ContentPage/ContentPage.tsx [hunk 2]
```diff
@@ -1,45 +1,41 @@
...
-type Props = {|
-  id: string,
-  isCreamy?: boolean,
-  Header: Element<typeof PageHeader>,
-  Body: Element<typeof Body>,
+type Props = {
+  id: string;
+  isCreamy?: boolean;
+  Header: ReactElement<typeof PageHeader>;
+  Body: any;
```
Comment (jamesgorrie) [inline common/views/components/ContentPage/ContentPage.tsx:31]: Why have we reverted to `any`?

File: common/views/components/ContentPage/ContentPage.tsx [hunk 3]
```diff
@@ -1,45 +1,41 @@
...
-  contributorProps?: ElementProps<typeof Contributors>,
-  Siblings?: Element<typeof SeriesNavigation>[],
-  outroProps?: ?ElementProps<typeof Outro>,
-  seasons?: any, // TODO
-|};
+  children?: ReactNode | null | undefined;
+  contributorProps?: typeof Contributors;
+  Siblings?: ReactElement<typeof SeriesNavigation>[];
+  outroProps?: ReactElement<typeof Outro> | null | undefined;
+  seasons?: any; // TODO
```
Comment (jamesgorrie) [inline common/views/components/ContentPage/ContentPage.tsx:37]: Could we fix this while we're at this?

File: common/views/components/ContentPage/ContentPage.tsx (hunk 2 above)
Comment (ishouty) [reply common/views/components/ContentPage/ContentPage.tsx:31]: Ahh my bad for this.. Thanks for checking

File: common/views/components/ContentPage/ContentPage.tsx (hunk 3 above)
Comment (ishouty) [reply common/views/components/ContentPage/ContentPage.tsx:37]: I could have a look at this.

File: common/views/components/ContentPage/ContentPage.tsx [hunk 4]
```diff
@@ -1,45 +1,47 @@
...
-  id: string,
-  isCreamy?: boolean,
-  Header: Element<typeof PageHeader>,
-  Body: Element<typeof Body>,
+type Props = {
+  id: string;
+  isCreamy?: boolean;
+  Header: ReactElement<typeof PageHeader>;
+  Body: {
+    props: {
```
Comment (jamesgorrie) [inline common/views/components/ContentPage/ContentPage.tsx:34]: This seems really un-standard, could you explain why we have to do this differently to everything else?

Comment (ishouty) [reply common/views/components/ContentPage/ContentPage.tsx:34]: Yes the Element is now not available from react in typescript so i had to change this. 
Function ShouldRenderBody() within the ContentPage,  looks for Body.props.body which does not exists , hence why i put this place. Any other suggestions?

Comment (jamesgorrie) [reply common/views/components/ContentPage/ContentPage.tsx]: @ishouty I've added [this](https://github.com/wellcomecollection/wellcomecollection.org/pull/6046/commits/4b420c696013df5521c06026e657f45cf16a7774). I imagine as we change over from flow, we'll need this a little more.

The problem was [this was the way to do this in flow](https://flow.org/en/docs/react/types/#toc-react-element).

File: common/views/components/Body/Body.tsx [hunk 5]
```diff
@@ -66,7 +71,7 @@ const LayoutWidth: FunctionComponent<LayoutWidthProps> = ({
   }
 };
 
-type Props = {
+export type Props = {
```
Comment (jamesgorrie) [inline common/views/components/Body/Body.tsx:74]:
```suggestion
type Props = {
```

**Review by jamesgorrie — APPROVED** [review]
Tidy

**Comment by ishouty** [comment]
nice one !

**Pull request merged**